vendor/deps/cli-ui/lib/cli/ui/os.rb

```diff
@@ -4,15 +4,17 @@ module OS
       # Determines which OS is currently running the UI, to make it easier to
       # adapt its behaviour to the features of the OS.
       def self.current
-        @current_os ||= case RUBY_PLATFORM
+        @current_os ||= case RbConfig::CONFIG['host_os']
         when /darwin/
           Mac
         when /linux/
           Linux
-        when /mingw32/
-          Windows
         else
-          raise "Could not determine OS from platform #{RUBY_PLATFORM}"
+          if RUBY_PLATFORM !~ /cygwin/ && ENV['OS'] == 'Windows_NT'
```

Comment:
We also need one for windows 10 pro x64 bit. The folder changes from `mingw32` to `mingw` on 64 bit based systems. This may only work for windows 11 users not windows 10 because some users will install a 32 bit version of RubyGems but be on a 64-bit OS.